secboot/encrypt_tpm.go

```diff
@@ -47,6 +48,31 @@ func NewRecoveryKey() (RecoveryKey, error) {
 	return key, err
 }
 
+func RecoveryKeyFromFile(recoveryKeyFile string) (*RecoveryKey, error) {
+	f, err := os.Open(recoveryKeyFile)
+	if err != nil {
+		return nil, fmt.Errorf("cannot open recovery key: %v", err)
+	}
+	defer f.Close()
+	st, err := f.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("cannot stat recovery key: %v", err)
+	}
+	if st.Size() != int64(len(RecoveryKey{})) {
+		return nil, fmt.Errorf("cannot read recovery key: unexpected size %v for the recovery key file", st.Size())
+	}
+
+	var rkey RecoveryKey
+	n, err := f.Read(rkey[:])
+	if err != nil {
+		return nil, fmt.Errorf("cannot read recovery key: %v", err)
+	}
+	if n != len(RecoveryKey{}) {
+		return nil, fmt.Errorf("cannot use recovery key: unexpected size %v", n)
+	}
```

Comment:
Can you not use io.ReadFull here, which avoids the need for the additional check on the number of bytes read?

Reply:
Yes, this is nicer indeed, thanks for this suggestion!